lib/svrquery/protocol/interfaces.go

```diff
@@ -15,6 +15,14 @@ type Responser interface {
 	MaxClients() int64
 }
 
+// ResponserV2 is an interface implemented by types which represent a query response.
+// 	It contains additional methods for advanced querying capabilities.
+type ResponserV2 interface {
+	Responser
+	TotalClientsConnectedEver() int64
+	NumBotClients() int64
+}
+
```

Comment:
While this works its not easily extensible / composable and also not clear without the description what it adds.

If you compare this to how `io` defines its interfaces with `Closer`, `Reader` etc.

The idiomatic approach is separate interfaces so that others can implement one without the other if needed.

This would be my suggestion, as you can see I've also tried to simplify the naming vs the specific protocol fields.
```suggestion
// Seener is implemented by types which can report the total number
// clients they have seen since start.
type Seener {
     SeenClients() uint64
}

// Botter is implemented by types which can report the number bots
// connected.
type Botter {
     Bots() uint64
}
```